text/0000-unkind-static-libraries.md

```diff
@@ -0,0 +1,54 @@
+- Feature Name: better_static_kind
+- Start Date: 2016-02-05
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Adds a new `kind=better_static` (name entirely open to bikeshedding) that is used to link static libraries by passing them to the linker, unlike `kind=dynamic` which is intended for dynamic libraries and `kind=static` which has rustc bundle it instead of passing it to the linker.
+
+# Motivation
+[motivation]: #motivation
+
+There is currently no way to properly link a static library by passing it to the linker instead of bundling it with rustc. `kind=dynamic` doesn't work because it thinks the library is a dynamic library resulting in issues such as passing the library on to later linker invocations beyond the first immediate linker invocation and also on Windows, once dllimport is actually supported correctly, `kind=dynamic` would cause dllimport to be emitted for all the symbols which is incorrect for static libraries. `kind=static` doesn't work because instead of passing the library to the linker, rustc bundles it, which results in rustc not looking in the standard library paths for the library.
```

Comment:
One other thing: this RFC is a golden opportunity to lay out the subtleties involved here (and there are many, it seems). It'd be really valuable to have a good exposition of the windows linking model and where Rust is falling short.